src/towncrier/test/test_create.py

```diff
@@ -151,20 +151,45 @@ def test_invalid_section(self):
             "Expected filename '123.foobar.rst' to be of format", result.output
         )
 
-    def test_file_exists(self):
+    @with_isolated_runner
+    def test_file_exists(self, runner: CliRunner):
         """Ensure we don't overwrite existing files."""
-        runner = CliRunner()
+        setup_simple_project()
+        frag_path = Path("foo", "newsfragments")
 
-        with runner.isolated_filesystem():
-            setup_simple_project()
+        for _ in range(3):
+            result = runner.invoke(_main, ["123.feature"])
+            self.assertEqual(result.exit_code, 0, result.output)
+
+        fragments = [f.name for f in frag_path.iterdir()]
+        self.assertEqual(
+            sorted(fragments),
+            [
+                "123.feature",
+                "123.feature.1",
+                "123.feature.2",
+            ],
+        )
 
-            self.assertEqual([], os.listdir("foo/newsfragments"))
+    @with_isolated_runner
+    def test_file_exists_with_ext(self, runner: CliRunner):
+        """Ensure we don't overwrite existing files."""
```

Comment:
this looks like a copy/paste error.

It would help to have a unique docstring that will explain why we have this tests, and what is expected from this test.